### PR TITLE
Preparing documentation for version 0.2 release 

### PR DESCRIPTION
**Pull request by waqasshabbir**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,43 +1,112 @@
...
+any string formats commonly found on web pages.
 
 
-* Free software: BSD license
-* Documentation: https://dateparser.readthedocs.org.
-
 Features
 --------
+* Generic parsing of dates in English, Spanish, Dutch, Russian and several other langauges and formats.
+* Generic parsing of contextual dates like: ``'1 min ago'``, ``'2 weeks ago'``, ``'3 months, 1 weeks and 1 day ago'``.
```
Comment (Allactaga) [inline README.rst]: `'3 months, 1 week and 1 day ago'` - no "s" in "weeks"

File: README.rst [hunk 2]
```diff
@@ -1,43 +1,112 @@
...
+* Generic parsing of dates with time zones abbreviations like: ``'August 14, 2015 EST'``, ``'July 4, 2013 PST'``.
+* Easy confingurations to change default parsing behavior. See :mod:`dateparser.conf`.
```
Comment (Allactaga) [inline README.rst]: Please move any configuration related documentation to separate branch. Configuration in fetus condition here and we do not want to encourage people to use it as it is because interface will change in the future. We are landing configuration in 0.3 release.

Comment (Allactaga) [inline README.rst]: I'll continue to review, after this change is done.

File: README.rst [hunk 3]
```diff
@@ -1,43 +1,111 @@
...
+* Generic parsing of dates with time zones abbreviations like: ``'August 14, 2015 EST'``, ``'July 4, 2013 PST'``.
+* Extensive test coverage.
+
+
+Usage
+-----
+The most straightforward way is to use the :func:`dateparser.parse` function,
+that wraps around most of the functionality in the module.
+
+.. automodule:: dateparser.__init__
```
Comment (Allactaga) [inline README.rst]: Do we need `__init__` here? It appears as `dateparser.__init__.parse` in compiled docs.

File: README.rst [hunk 4]
```diff
@@ -1,43 +1,111 @@
...
+    datetime.datetime(2014, 12, 11, 9, 0)
+    >>> dateparser.parse(u'13 января 2015 г. в 13:34')  # Russian
+    datetime.datetime(2015, 1, 13, 13, 34)
+    >>> dateparser.parse(u'1 เดือนตุลาคม 2005, 1:00 AM')  # Thai
+    datetime.datetime(2005, 10, 1, 1, 0)
+
+This will try to parse a date from the given string, attempting to
+detect the language each time.
+
+If you know beforehand the languages that the date should be, you can skip
```
Comment (Allactaga) [inline README.rst]: Is it grammatically correct sentence?

File: README.rst [hunk 5]
```diff
@@ -1,43 +1,111 @@
...
-* it can give you the date for text like: ``'1 min ago'``, ``'2 weeks ago'``, ``'3 months, 1 weeks and 1 day ago'``, etc
+If you know the possible formats that the date will be, you can
+use the ``date_formats`` argument::
 
+    >>> dateparser.parse(u'22 Décembre 2010', date_formats=['%d %B %Y'])
+    datetime.datetime(2010, 12, 22, 0, 0)
 
-The goal is to support the common date formats used in websites all around the world.
+
+Contextual Dates
```
Comment (Allactaga) [inline README.rst]: Do we call them that? Aren't they "relative"? :)

File: README.rst [hunk 6]
```diff
@@ -1,43 +1,111 @@
...
+    datetime.datetime(2012, 12, 21, 1, 0)
+    >>> parse(u'Hace una semana')  # Spanish
+    datetime.datetime(2012, 12, 15, 0, 0)
+    >>> parse(u'2小时前')  # Chinese
+    datetime.datetime(2012, 12, 21, 22, 0)
+
+.. note:: Testing above code might return different values for you depending on your environment's current date time. To reproduce exactly above results, you can patch your environment as follows::
+
+    >>> from dateparser.date import freshness_date_parser
+    >>> patch.object(freshness_date_parser, 'now', datetime(2012, 12, 22, 0, 0)).start()
```
Comment (Allactaga) [inline README.rst]: As strange as it sounds, not everybody can be familiar with mocking mechanism. Let's just use more recent date (like 1st June 2015) and state that user would get different results depending on the current date and time. Also let's provide translations for the given dates (f.ex "French (2 hours ago)"). And "yaklaşık 23 saat önce" is not russian :smile:

Comment (waqasshabbir) [inline README.rst]: Oh yes, it's turkish :facepalm: :D

File: dateparser/date.py [hunk 7]
```diff
@@ -228,6 +228,16 @@ def _is_valid_date_obj(self, date_obj):
...
+        """
+        :param languages:
+                A list of two letters language codes.e.g. ['en', 'es'].
+                If languages are given, it will not attempt to detect the language.
+
+        :param allow_redetect_language:
+                Enables/disables language re-detection.
+        
+        :return: A parser instance
+        """
```
Comment (Allactaga) [inline dateparser/date.py]: Hm. Where is this docstring appears on the docs?

File: dateparser/date.py [hunk 8]
```diff
@@ -248,10 +258,26 @@ def __init__(self, languages=None, allow_redetect_language=False):
...
+
+        .. _here: https://docs.python.org/2/library/datetime.html#strftime-and-strptime-behavior
+            :param date_formats:
+
+        :return: a dictionary with a :mod:`datetime.datetime` object and a *period*.
+        .. note:: *Period* values can be a 'day' (default), 'week', 'month', 'year'.
+
+        *Period* aims to solve the following issue:
+
+        For example, a forum could displays "2 weeks ago" in the thread list
```
Comment (Allactaga) [inline dateparser/date.py]: Is this paragraph relevant to dateparser functionality? Especially "won’t be scraped" part?

File: docs/usage.rst [hunk 9]
```diff
@@ -1,82 +1,122 @@
...
-    dateparser.date_parser.LanguageWasNotSeenBeforeError
+    >>> ddp = DateDataParser(languages=['de', 'nl'])
+    >>> ddp.get_date_data(u'vr jan 24, 2014 12:49')
+    {'date_obj': datetime.datetime(2014, 1, 24, 12, 49), 'period': u'day'}
+    >>> ddp.get_date_data(u'18.10.14 um 22:56 Uhr')
+    {'date_obj': datetime.datetime(2014, 10, 18, 22, 56), 'period': u'day'}
 
 
-If you want it to redetect the language every time, you can use a custom date_parser, like so::
+Deploying dateparser in a Scrapy Cloud project
```
Comment (Allactaga) [inline docs/usage.rst]: Would not it be a better fit for 'Installation' doc?

File: dateparser/date.py (hunk 7 above)
Comment (waqasshabbir) [inline dateparser/date.py]: I think it doesn't appear anywhere. Any suggestions, where to put this docstring if it makes sense using it?

File: docs/installation.rst [hunk 10]
```diff
@@ -4,9 +4,17 @@ Installation
...
-Or, if you have virtualenvwrapper installed::
+If you want to install from the latest sources, you can do::
 
-    $ mkvirtualenv dateparser
-    $ pip install dateparser
+    $ git clone https://github.com/scrapinghub/dateparser.git
+    $ cd dateparser
+    $ python setup.py install
+
+Depending on how your environment is set up, you might need to use sudo for the
```
Comment (Allactaga) [inline docs/installation.rst]: Let's not recommend sudo over creating virtual environment. I think our users are smart enough to figure differences on their own, and it would not be the first  module they install :)

File: dateparser/date.py (hunk 8 above)
Comment (waqasshabbir) [inline dateparser/date.py]: Not sure, it was there already so left it as is.

File: docs/usage.rst (hunk 9 above)
Comment (waqasshabbir) [inline docs/usage.rst]: Agreed.

File: dateparser/date.py (hunk 8 above)
Comment (Allactaga) [inline dateparser/date.py]: I think it's relevant more for bot_engines and not for dateparser. Though we need to explain period to the users more. In a sense it's a margin of error on how we calculate a data. May be give few more examples here (like for 'year'/'month' periods). Explain the uncertainty of "2 weeks ago" date.

**Comment by Allactaga** [comment]
It is not required for this release, but I think we can write few more details about forking/pull-requesting back in `Contributing`/`Get Started!` to make it more easy for those who have not done it before.

**Comment by Allactaga** [comment]
Also in `contirbuting`/`Pull Request Guidelines` I think we can add something like:
4. Follow the core developers' advices as they are given for keeping code consistent regardless of variety approaches used by many contributors.
5. If you don't have time to continue working on pull request, or simple lost interest. Please write a 5 second comment about it, so we know we can apply changes there to get it done.

Or something similar in meaning.

**Comment by Allactaga** [comment]
And do we need `Contributing`/`Tips` with one tip about `nosetests` speed up? We run whole suite in 5 seconds and this is well known approach for anyone uses `nose` and we kinda "tipping" to list folder contents with `ls` command.

**Comment by Allactaga** [comment]
As for `Credits` I think everybody deserve to be there.
Here is the list of all commiters:
```
Artur Sadurski
Claudio Salazar
Elias Dorneles
Eugene Amirov
Faisal Anees
Ismael Carnales
Joseph Kahn
Mark Baas
Marko Horvatić
Mateusz Golewski
Opp Lieamsiriwong
Rajat Goyal
Raul Gallegos
Shuai Lin
Sviatoslav Sydorenko
Tom Russell
Umair Ashraf
Waqas Shabbir
```
I also added  Ismael and Shuai here as they were at the beginning of this module, but moving it to separate repository broke commit history.

File: dateparser/date.py (hunk 7 above)
Comment (eliasdorneles) [inline dateparser/date.py]: Hey fellows! :)

The docstring appear in the docs, they're just not referenced much in the main docs text -- they're collected by Sphinx `autodocs` extension, enabled in `docs/conf.py`.

If you go to the module index: http://dateparser.readthedocs.org/en/latest/py-modindex.html
you can click in a module and it'll take you to that module's docs: http://dateparser.readthedocs.org/en/latest/dateparser.html

**Comment by eliasdorneles** [comment]
+1 on the `Credits` list!
At the time I extracted the package, it was a pain to get the commiters from git -- the history wasn't reliable because I basically copied the code from the internal project.
Sorry for any injustices! ;)

File: docs/installation.rst (hunk 10 above)
Comment (eliasdorneles) [inline docs/installation.rst]: Hmm, not sure. Users that are smart enough to figure out on their own probably don't need these docs, I was thinking it's nice to assume this is the first module they're installing. ;)

File: docs/usage.rst (hunk 9 above)
Comment (eliasdorneles) [inline docs/usage.rst]: +1

File: README.rst (hunk 5 above)
Comment (eliasdorneles) [inline README.rst]: +1 for "relative".

File: README.rst [hunk 11]
```diff
@@ -1,43 +1,107 @@
...
+    >>> dateparser.parse(u'1 เดือนตุลาคม 2005, 1:00 AM')  # Thai
+    datetime.datetime(2005, 10, 1, 1, 0)
+
+This will try to parse a date from the given string, attempting to
+detect the language each time.
 
-If you have needed to parse dates before, you probably have used the date
-parser of the dateutil_ module.
-We built this library on top of it, adding a few features:
+You can specify the language(s), if known, using ``languages`` argument. In this case, given languages, if supported, are used and language detection is skipped::
```
Comment (Allactaga) [inline README.rst]: We raise an error if language is unknown, don't we? Do we need "if supported part" here?

File: dateparser/date.py [hunk 12]
```diff
@@ -248,17 +258,39 @@ def __init__(self, languages=None, allow_redetect_language=False):
...
+        *Period* represent the granularity of date parsed from the given string.
+
+        In the example below, since no day information is present, the day is assumed to be current
+        day ``16`` from *June 16, 2015*. Hence, the level of precision is ``month``. 
+
+            >>> DateDataParser().get_date_data(u'March 2015')
+            {'date_obj': datetime.datetime(2015, 3, 16, 0, 0), 'period': u'month'}
+
+        Similarly, for date strings with no day and month information present, level of precision
+        is ``year`` and day ``16`` and month ``6`` are from *June 16, 2015*.
```
Comment (Allactaga) [inline dateparser/date.py]: ...from current date (which is June 16, 2015 at the moment of writing this)
